sysvar/src/lib.rs

```diff
@@ -161,6 +161,27 @@ pub trait SysvarSerialize:
 /// Implements the [`Sysvar::get`] method for both SBF and host targets.
 #[macro_export]
 macro_rules! impl_sysvar_get {
+    () => {
+        fn get() -> Result<Self, $crate::__private::ProgramError> {
+            // Allocate uninitialized memory for the sysvar struct
+            let mut uninit = core::mem::MaybeUninit::<Self>::uninit();
+            let size = core::mem::size_of::<Self>() as u64;
+            // Safety: we build a mutable slice pointing to the uninitialized
+            // buffer.  The `get_sysvar` syscall will fill exactly `size`
+            // bytes, after which the buffer is fully initialised.
+            let dst = unsafe {
+                core::slice::from_raw_parts_mut(uninit.as_mut_ptr() as *mut u8, size as usize)
+            };
+            // Attempt to load the sysvar data.
+            $crate::get_sysvar(dst, &id(), 0, size)?;
```

Comment:
nit: assuming that `id()` is available (and correct) in the scope of the macro could cause unexpected errors -- how about passing the sysvar id explicitly in the macro definition?

that way callers would call the macro like `impl_sysvar_get(id())` instead

Reply:
Oof, good note. Brittle, and tests overfitted.

Passing as you suggest is clearly the best option, as opposed to specifying/checking in macro. Done in 57bf78e5